feat: made an increment on slidedeck (#1)
mftenergy/Gitting-comfortable: slidedeck/slides.md

mdc: true
---

# Welcome to *Git*ting comfortable

A short introduction to Git

<div class="pt-12">
  <span @click="$slidev.nav.next" class="px-2 py-1 rounded cursor-pointer" hover="bg-white bg-opacity-10">
    Press Space or use keyboard arrows for next page <carbon:arrow-right class="inline"/>
  </span>
</div>

---

# Basics of Git

Git is a version control system that helps you manage text files of all sorts in a collaborative and structured manner. <br>

The concept of git is that you work with a `repository` of which you can create `branches` (think shortlived variants of the codebase) that enables you to collaborate from the same point in time. You `commit` changes to a branch to create a record of your change for others and yourself to keep track of the history in changes. <br>

At the end you sync changes to a git server with `push` in order to save work and synchronize changes with your colleagues or automation systems.

```bash

git add <file> # Add a file to "staging"

git commit -m <message> # Records all changes that are staged by add with documentation on the work done in the files

> git push -u origin <branch> # Push all missing commits from localhost to a git server syncing your records with a server for collaboration

Reviewer: ```suggestion
git push -u origin <branch> # Push all missing commits from localhost (your machine) to a git server syncing your records with a server for collaboration
```

git checkout -b <branch> # Create and switch branch in a single command
```

---
layout: two-cols
---

# Basics of Git: Client-server

Git works as a client server paradigm 
<br>
<br>

> Git works by a client (a user) downloads the source code from a `repository`, makes edit locally and syncs these edits to `origin` (the server of which you downloaded the source code). <br><br>

Reviewer: ```suggestion
Git works by a client (a user) cloning the source code from a `repository`, performing edits locally and pushing these edits to `origin` (the server from which you downloaded the source code). <br><br>
```

</v-click>

<v-click at=5>
```mermaid
---
title: Automation
---
flowchart LR
  id1(checkout code)
  id2(run quality check)
  id3(deploy service)
  
  id1 --> id2
  id2 --> id3
```
</v-click>
---

# Working with branches

TODO; Talk about branching

- Checkout from main
- Do edits
- Commits
- Push
- PR

---

# Working with merges / rebasing

TODO; Talk about merge and rebasing

- In relation to PRs
- In relation to local, where we prefer rebase

---

# Reverting / Rest

TODO; When shit hits the fan

- Revert
- Reset

---

# Stashing changes

TODO; Stashing

---

# Time to do exercises!

- In the repo; https://github.com/mftenergy/Gitting-comfortable there are exercises
- Follow the path explained at https://github.com/mftenergy/Gitting-comfortable?tab=readme-ov-file#suggested-learning-path
- Path into an exercise directory, run `setup.sh` and read the README.md for instruction on what to do

```bash
cd basic-commits
./setup.sh
cat README.md
git magic
```

NOTE: There is a cheatsheet at the bottom of the README.md in the root of the repository.

---

> # What is Slidev? asdasd test

Reviewer: ```suggestion
```

Author: Keep for now to make inspiration when you develop slides

> Slidev is a slides maker and presenter designed for developers, consist of the following features

Reviewer: ```suggestion
```
